### PR TITLE
[WOR-1284] Switch Azure e2e test to using the async deletion API

### PR DESCRIPTION
**Pull request by cahrens**
Ticket: https://broadworkbench.atlassian.net/browse/WOR-1284

Successful Azure e2e test run on this branch: https://github.com/broadinstitute/rawls/actions/runs/6659234257

---

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

### CODE REVIEW

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspacesAzureApiSpec.scala [hunk 1]
```diff
@@ -157,12 +157,16 @@ class WorkspacesAzureApiSpec extends AnyFlatSpec with Matchers with BeforeAndAft
           verifyBlobNotCloned(cloneSasUrl, nonAnalysesFilename)
         }
       } finally {
-        Rawls.workspaces.delete(projectName, workspaceCloneName)
-        assertNoAccessToWorkspace(projectName, workspaceCloneName)
+        withClue(s"deleting the cloned workspace ${workspaceCloneName} failed") {
```
Comment (cahrens) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspacesAzureApiSpec.scala:160]: In one of the iterations, I got a failure from deleting one of these workspaces, and it was unclear to me which one it was.

**Comment by cahrens** [comment]
jenkins retest

**Review by marctalbott — APPROVED** [review]

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
jenkins retest

**Review by blakery — APPROVED** [review]

**Pull request merged**